### DISCUSSION TITLE
Cannot stringify arbitrary non-POJOs - Prisma w/ SqlServer

### DISCUSSION THREAD

**Discussion by bmccorm2** [prisma/prisma#19063] [Q&A]
Hi - I have prisma hooked up to an existing sqlServer instance and trying to bring back a decimal(32,16) field - coupon.  When i try to return this in my response object, i get the error above.  Could anyone shed light on why it is having trouble with decimal fields?

```
  let test = await prisma.security_dim.findMany({
    where: {
      SECURITY_SCHEME: 'finc'
    },
    select: {
      name: true,
      short_name: true,
      coupon: true
    },
    take: 5
  })
```

**Comment by nurul3101** [comment]
Hello @bmccorm2 👋 

Which version of Prisma are you using?
Can you share how you have defined the Decimal field in your model? I can try to reproduce this on my end and check.

**Comment by bmccorm2** [comment]
I am using 4.13 Prisma.  SqlServer i am using a docker image: `mcr.microsoft.com/mssql/server:2019-latest`

I have an existing db so i used introspection to create this migration script. (there is over 200 columns so i've shortened it substantially :) 
```
CREATE TABLE [dbo].[security_dim] (
    [securityKey] INT NOT NULL,
    [name] VARCHAR(1000),
    [shortName] VARCHAR(1000),
    [SECURITY_SCHEME] VARCHAR(1000),
   ........
    [coupon] DECIMAL(32,16),
)
```

**Reply by nurul3101** [reply]
My hunch is that this could have something to do with the response returning Decimal values as instances of the [Decimal.js](https://github.com/MikeMcl/decimal.js/) library, which would not be serialized. 
You get this error as soon as you invoke this function, right?
```
  let test = await prisma.security_dim.findMany({
    where: {
      SECURITY_SCHEME: 'finc'
    },
    select: {
      name: true,
      short_name: true,
      coupon: true
    },
    take: 5
  })
```

**Comment by bmccorm2** [comment]
(I'm using svelteKit) - No - I get the error as soon as i return the response object.  I think you might be right though - if i create a test object like this and `console.log` it, the decimal portion is color coded yellow.  However, when i `console.log` the response from `test` above, the decimal portion is NOT color coded.

**WORKS**
```
  let test = {
    name: 'bryan',
    short_name: 'bryan',
    coupon: 4.356
  }
```

**DOES NOT WORK**
```
  let test = await prisma.security_dim.findUnique({
    where: {
      securityKey: 1
    },
    select: {
      name: true,
      short_name: true,
      coupon: true
    }
  })
```

**Reply by nurul3101** [reply]
Interesting! Is your repository open source so that I can test it?

**Comment by bmccorm2** [comment]
Sorry i can't open source it because it is for a company :(  Here are the two relevant files though.  Just any column that is a decimal I have trouble with.

**+page.server.ts**  (loads data for the page)
```
import prisma from "$lib/server/prisma";

export const load = async ({ params }) => {
  const securityKey = Number(params.id)

  //DOES NOT WORK
  const test = await prisma.security_dim.findUnique({
    where: {
      SECURITY_KEY: securityKey
    },
    select: {
      BLOOMBERG_TICKER: true,
      SECURITY_SHORT_NAME: true,
      COUPON_RATE: true
    }
  })

  //WORKS
  // const test = {
  //   BLOOMBERG_TICKER: 'TEST',
  //   SECURITY_SHORT_NAME: 'NAME',
  //   COUPON_RATE: 5.3750000000000000  //value in the DB
  // }

  return {
    test
  }
};
```
```
Error: Data returned from `load` while rendering /test/[id] is not serializable: Cannot stringify arbitrary non-POJOs (data.test.COUPON_RATE)
```
![image](https://github.com/prisma/prisma/assets/11781610/4e3f0008-8658-4d39-af73-43f63a0012c8)

Below, notice how VS code knows it is a number so it highlights at as such vs above it doesn't know the data type
![image](https://github.com/prisma/prisma/assets/11781610/81f6c293-145c-4f4f-b7f3-91572db2edf4)


**DB Table Creations script**  (from introspection)
```
create table security_dim (
	SECURITY_KEY int primary key,
	SECURITY_SHORT_NAME VARCHAR(1000),	
	BLOOMBERG_TICKER VARCHAR(1000),
	COUPON_RATE decimal(32,16),
);
```

**Reply by nurul3101** [reply]
Thanks for sharing the files. Are you using any specific sveltekit template? If yes, I could use the same one to try to replicate this locally.

**Reply by nurul3101** [reply]
Checking to see if you had a chance to look at my last response.

**Reply by bmccorm2** [reply]
Sorry was out of the country.  No special templates - just the blank template from creating a new app (npm create svelte@latest) and then i add the prisma packages.

**Comment by ieedan** [comment]
I get the same error trying to send this response from prisma from my load function.

```json
[
  {
    id: 'a2724d1a-2cb7-4e16-abad-b9d42c985aab',
    name: 'Mouse',
    description: 'Mousey mouse',
    price: '12.9'
  }
]
```

**"Error: Data returned from `load` while rendering / is not serializable: Cannot stringify arbitrary non-POJOs (data.products[0].price)"**

Code:

```ts
export async function load({ setHeaders }) {
    let products: Product[] = [];
    try {
        products = await prisma.product.findMany();        
    } catch (error) {
        console.log(error);        
    }

    return {
        products
    };
}
```

This isn't a great solution but if you stringify then parse the result from the client svelte will allow you to send the response.

Something like this 

```ts
export async function load({ setHeaders }) {
    let products: Product[] = [];
    try {
        products = await prisma.product.findMany();        
    } catch (error) {
        console.log(error);        
    }

    // stringify then parse the result from prisma
    const json = JSON.stringify(products);
    const jsonResult = JSON.parse(json);

    return {
        products: jsonResult
    };
}
```

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that haven’t had any recent activity and appear to be outdated.

If this discussion is still important to you or unresolved, we’d love to hear from you! Feel free to reopen it or start a new one with updated details.

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and being part of the community!
